cpp/src/io/parquet/rle_stream.cuh

```diff
@@ -56,92 +55,102 @@ inline __device__ uint32_t get_vlq32(uint8_t const*& cur, uint8_t const* end)
   return v;
 }
 
-// an individual batch. processed by a warp.
-// batches should be in shared memory.
-template <typename level_t>
-struct rle_batch {
-  uint8_t const* run_start;  // start of the run we are part of
-  int run_offset;            // value offset of this batch from the start of the run
-  level_t* output;
-  int level_run;
-  int size;
-
-  __device__ inline void decode(uint8_t const* const end, int level_bits, int lane, int warp_id)
-  {
-    int output_pos = 0;
-    int remain     = size;
+template <typename level_t, int max_output_values>
```

Comment:
Docstring for this function.  Lots of parameters here that sound very similar.

Reply:
Pushed a change here with a header: https://github.com/rapidsai/cudf/pull/15159/commits/bb39e9fa9144079906168503fcd046d16d8b7065. I also grouped the arguments in a hopefully more logical way.